Update supported versions in the `bin/check-dependencies` script

YevhenZvieriev opened the pull request:
Magento 2.4.7 has been released, so I removed the dependencies for such versions as `2.4.7-beta3` and `2.4.7-beta2` from the `bin/check-dependencies` script and added `2.4.7` and `2.4.6-p5` versions.

![Screenshot from 2024-04-09 22-08-01](https://github.com/markshust/docker-magento/assets/43544955/bbae533c-3b97-4497-ae56-df7f722a4e86)

markshust commented:
Thanks! Note that there are also new versions `2.4.5-p7` and `2.4.4-p8`. Do these versions need related updates?

YevhenZvieriev commented:
Good point! I'll add the system requirements of the new versions in the next commit.

YevhenZvieriev commented:
I added system requirements for new versions of Magento 2 so now the script provides system requirements for all supported versions.

Additionally, I added a link to the Magento system requirements documentation from DevDocs if the version entered by the user is not found in the list. This enhancement provides users with easy access to detailed information about the system requirements for running Magento, facilitating smoother installation and setup processes.

![Screenshot from 2024-04-13 13-30-43](https://github.com/markshust/docker-magento/assets/43544955/76575c49-3515-4aaa-893d-105d41431504)
